app/grandchallenge/reader_studies/forms.py

```diff
@@ -299,6 +312,20 @@ def __init__(self, *args, **kwargs):
             )
         )
 
+    def save(self, *args, **kwargs):
+        instance = super().save(*args, **kwargs)
+        if "widget" in self.changed_data:
+            new_widget = self.cleaned_data["widget"]
+            try:
+                old_widget = instance.widget
+            except ObjectDoesNotExist:
+                old_widget = None
+            if old_widget.pk:
+                old_widget.delete()
```

Comment:
`if old_widget.pk:` won't this error out if `old_widget = None`? Maybe this is what is intended?

```suggestion
            try:
                instance.widget.delete()
            except ObjectDoesNotExist:
                pass
```